### DISCUSSION TITLE
Welcome to RookieAI_yolov8 Discussions!

### DISCUSSION THREAD

**Discussion by Passer1072** [Passer1072/RookieAI_yolov8#11] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 321623su** [comment]
How to make changes to the yolov8 code structure or replace the module, where is the tutorial

**Comment by Passer1072** [comment]
> How to make changes to the yolov8 code structure or replace the module, where is the tutorial

If you are looking to modify the inference part of yolov8, you can find more resources on the Ultralytics website. For changes specific to our project code, there is currently no tutorial available, but please stay tuned for future updates.

**Comment by AAAYNMMM** [comment]
当在游戏内时程序的fps会降低到1，使用截图测试倒是正常，并且鼠标能够移动。有解决办法吗？（彩6已开启无边框模式）
![屏幕截图 2024-06-24 171727](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/557bb091-fc2c-48ed-b57e-4ccf4d77208c)

**Reply by Passer1072** [reply]
> 当在游戏内时程序的fps会降低到1，使用截图测试倒是正常，并且鼠标能够移动。有解决办法吗？（彩6已开启无边框模式）
> 
> ![屏幕截图 2024-06-24 171727](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/557bb091-fc2c-48ed-b57e-4ccf4d77208c)
> 
> 

游戏内锁帧了吗？，降低画质，锁帧给ai留下算力。

**Reply by AAAYNMMM** [reply]
你好，apex也是如此。游戏没锁帧，尝试过锁帧，并无效果，我再去试验一番

**Reply by AAAYNMMM** [reply]
测试有效，截图fps正常。但是游戏鼠标却无法移动，有解决方法吗

**Reply by Passer1072** [reply]
> 测试有效，截图fps正常。但是游戏鼠标却无法移动，有解决方法吗

2.4版本需要配合飞易来U盘使用，2.3版本不需要

**Reply by AAAYNMMM** [reply]
使用的就是2.3.2版本

**Comment by AAAYNMMM** [comment]
你好，我已找到方法可以移动鼠标，调用autohotkey即可

**Reply by AAAYNMMM** [reply]
调用半天鼠标乱飞未能解决

**Reply by Passer1072** [reply]
> 你好，我已找到方法可以移动鼠标，调用autohotkey即可

鼠标移动库已回滚为win32库，应该可以使用了。

**Reply by AAAYNMMM** [reply]
可以使用了。但是画面太抖（彩虹6号），自瞄一顿一顿（apex），不如手打。还有很大的优化空间，建议在中间加一个小圈，如果这个小圈覆盖住了就不再自瞄，中心点出圈后启动自瞄立马拉到中心点，这样有一个范围缓冲，打微移动的角色不会抖来抖去。以上过于主观，因使用环境配置模型的不同会有偏差

**Reply by Passer1072** [reply]
> 可以使用了。但是画面太抖（彩虹6号），自瞄一顿一顿（apex），不如手打。还有很大的优化空间，建议在中间加一个小圈，如果这个小圈覆盖住了就不再自瞄，中心点出圈后启动自瞄立马拉到中心点，这样有一个范围缓冲，打微移动的角色不会抖来抖去。以上过于主观，因使用环境配置模型的不同会有偏差

自瞄一顿一顿是因为显卡占用高导致推理延迟高推理帧数低导致的，只能通过降低游戏的显卡占用来优化，例如开到最低画质，限制游戏帧数。我4080，apex开最低画质2K推理帧数能到70帧就很流畅，左右晃也是因为推理延迟高导致的还没有找到合适的解决办法。
重装一个干净的游戏系统也可以提高推理帧数，我装了一个Atlas windows，主系统推理帧数只有30，在这个系统中可以达到60+。这就是个精简版windows可以尽可能减少外部资源占用。

**Reply by AAAYNMMM** [reply]
![屏幕截图 2024-06-27 121245](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/01db6e2d-813a-4d84-b577-b4bef85fd581)
把强锁范围关闭可以解决这个问题，画面不再抖动，不会出现抖出人物范围的情况

**Reply by AAAYNMMM** [reply]
另外建议加一个睡眠，免得一直刷在加载
![屏幕截图 2024-06-27 121731](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/8a7baedd-6f54-497e-89c7-77e1bc4c2ace)
另外希望可以把控制台的刷新关掉
![屏幕截图 2024-06-27 121824](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/c0a9018c-4174-4688-a9ae-2be6473409df)
目前这三个不知道在哪关闭
![image](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/eef81b2b-bb0c-4d79-8b4e-5f1c1bf68232)

**Reply by Passer1072** [reply]
> 另外建议加一个睡眠，免得一直刷在加载
> 
> ![屏幕截图 2024-06-27 121731](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/8a7baedd-6f54-497e-89c7-77e1bc4c2ace)
> 
> 另外希望可以把控制台的刷新关掉
> 
> ![屏幕截图 2024-06-27 121824](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/c0a9018c-4174-4688-a9ae-2be6473409df)
> 
> 目前这三个不知道在哪关闭
> 
> ![image](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/eef81b2b-bb0c-4d79-8b4e-5f1c1bf68232)
> 
> 

如果软件测试没有问题可以在打包的时候选择不打包控制台，详细参数刷屏在之后有一个版本开始已经默认关闭了，可能没有同步到正式版中，详见这个问答https://github.com/Passer1072/RookieAI_yolov8/issues/13

**Comment by AAAYNMMM** [comment]
这个功能需要开启才能瞄准，不然会偏离（原因未知）
![屏幕截图 2024-06-27 150522](https://github.com/Passer1072/RookieAI_yolov8/assets/96669024/12d0414c-2a21-4f78-91d0-1ed10066501e)
另外上面配置的截图还可以再优化优化，强锁范围速度都设置0，倍率可以设置在0.3，启用倍率预测。（不过貌似关闭倍率和不预测没差）
另外请问有没有训练模型的教程可以分享一下链接，yolov5或者此自瞄的模型版本都行

**Reply by Passer1072** [reply]
> 这个功能需要开启才能瞄准，不然会偏离（原因未知） ![屏幕截图 2024-06-27 150522](https://private-user-images.githubusercontent.com/96669024/343666362-12d0414c-2a21-4f78-91d0-1ed10066501e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1Mzg1MzcsIm5iZiI6MTcxOTUzODIzNywicGF0aCI6Ii85NjY2OTAyNC8zNDM2NjYzNjItMTJkMDQxNGMtMmEyMS00Zjc4LTkxZDAtMWVkMTAwNjY1MDFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI4VDAxMzAzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0MWU1MzJlZTNlNTQ1M2FlMzdlMmYyNWFmY2JkMTkwZDExOTMzMDViNzg4NmI4YjhmZjg3YjQwOTJiYWZmYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qQVe-tTaMMuPfEv7e0cxQlIyYLUDdhdNsCxl0vYWoR4) 另外上面配置的截图还可以再优化优化，强锁范围速度都设置0，倍率可以设置在0.3，启用倍率预测。（不过貌似关闭倍率和不预测没差） 另外请问有没有训练模型的教程可以分享一下链接，yolov5或者此自瞄的模型版本都行

倍率预测算是一个废案，最初目的是实现自动预判位置，但是后来发现并不好实现，该功能目前主观使用效果体现在加快瞄准速度一般用在分段瞄准中强锁范围内让强锁力度更强。这个功能之后会做一些更改。
如果强锁范围设置为0其实可以考虑直接关闭分段瞄准模式直接使用普通瞄准模式。
关于训练教程这里我参考的是[YOLOv5训练](https://www.youtube.com/watch?v=2TIhdai0f6M&list=PL2Ayzx9o5SMTC2bj4gj1qm-uvf9FjutW5)但是软件是针对YOLOv8适配，这个教程也适用。

**Comment by molanp** [comment]
## 关于全局变量

>  在python函数中，如果使用了全局变量，但并没有对其进行赋值（类操作不算赋值），那么不需要使用`global`字段声明该变量


所以说，我感觉这个函数引用还可以再精简，计划尝试使用二分法删减进行测试，以实现仅保留必要声明

**Comment by lasyxina** [comment]
有新版成品可以下载吗

**Reply by Passer1072** [reply]
> 有新版成品可以下载吗

最新Beta版较大幅度更改了分段瞄准逻辑，添加了三四个新参数并且还没有做UI适配，只能通过代码调整参数，目前正在测试这个逻辑的可行性，如果效果不错才会代替现有逻辑并进行UI的适配。

**Reply by lasyxina** [reply]
因为我是小白不会弄这个东西难过

**Reply by molanp** [reply]
> 因为我是小白不会弄这个东西难过

- 下载python，建议3.10，然后百度`pip换源阿里`
- 复制readme里面的3.10部分安装依赖代码到终端
-  回车
- 把项目下载到本地，在文件夹中打开cmd/终端/powershell（管理员）
-  输入 `python xxx.py`，其中xxx为项目里那个rookie开头的文件名
- 回车启动

**Reply by lasyxina** [reply]
可能是我太笨的原因我弄完了也是不好用打不开 可不可以给我发一份成品呢

**Reply by lasyxina** [reply]
332848588@qq.com

**Comment by PZay05** [comment]
如何使用2.5.6版本，想用onnx模型怎么搞 请教一下 大佬

Traceback (most recent call last):
  File "E:\ai\RookieAI_yolov8-main\Beta\V2.5.0(beta)\RookieAI_YOLOv8_V2.5.6.py", line 130, in read
    with open("settings.json", "r", encoding='utf-8') as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'settings.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "E:\ai\RookieAI_yolov8-main\Beta\V2.5.0(beta)\RookieAI_YOLOv8_V2.5.6.py", line 159, in <module>
    Opt = Option()
          ^^^^^^^^
  File "E:\ai\RookieAI_yolov8-main\Beta\V2.5.0(beta)\RookieAI_YOLOv8_V2.5.6.py", line 85, in __init__
    self.content = self.read()
                   ^^^^^^^^^^^
  File "E:\ai\RookieAI_yolov8-main\Beta\V2.5.0(beta)\RookieAI_YOLOv8_V2.5.6.py", line 133, in read
    return self.default
           ^^^^^^^^^^^^
AttributeError: 'Option' object has no attribute 'default'

**Reply by molanp** [reply]
已修复
544f622e78ff3759ae0fff3731d9e3c1322ba111

重新下载文件即可

**Comment by PZay05** [comment]
如果我想使用onnx的模型用cuda12.1版的可以？  一定要用cudnn9.3.0版本？

**Reply by molanp** [reply]
cudnn需要和cuda版本匹配，不要求9.3.0版本
